syntax = "proto3";

// Bound represents a limit for the performance value
message Bound {
    double bound = 1;
}

// Criteria captures a single pass criteria
message Criteria {
    // name is presented as a regular expression
    string name_regex = 1;

    // only apply criteria to this test, if populated
    string test_name = 2;

    // lower and upper Bound messages as describe above
    Bound lower_bound = 3;
    Bound upper_bound = 4;
}

// PassCriteria contains any number of criteria as above
message PassCriteria {
    repeated Criteria criteria = 1;
}